Sources/Moya/MoyaProvider+Internal.swift

```diff
@@ -83,18 +84,25 @@ public extension MoyaProvider {
             switch stubBehavior {
             case .never:
                 switch target.task {
```

Comment:
A little trivial and unrelated to the PR - but can we pull this switch statement out into a `sendRequest` function? This way we don't have nested switch statements and can start cleaning up this behemoth of a method 😃